app/backend/src/couchers/servicers/account.py

```diff
@@ -117,12 +119,19 @@ def ChangeEmail(self, request, context):
 
         with session_scope() as session:
             user = session.query(User).filter(User.id == context.user_id).one()
-
-            # otherwise we're good
             user.new_email = request.new_email
-            token, expiry_text = set_email_change_token(session, user)
 
-            send_email_changed_notification_email(user)
-            send_email_changed_confirmation_email(user, token, expiry_text)
+            if user.has_password:
+                old_email_token, new_email_token, expiry_text = set_email_change_tokens(
+                    session, user, double_confirmation=False
+                )
+                send_email_changed_notification_email(user)
+            else:
+                old_email_token, new_email_token, expiry_text = set_email_change_tokens(
```

Comment:
Please add a comment here about how this works.

It's also probably cleaner to move the `send_email_changed_confirmation_email(...)` inside the branches instead of leaving it out, just makes it clearer how the logic works, even though it's an extra line

Reply:
There's a comment section under the method name (which needed some updating). Did you mean something different? Or should I move that comment down here?